### PR TITLE
add the opportunity configure "binary_path" from config

### PR DESCRIPTION
**Pull request by fatonsopa**
added: binary_path on config. (works only with the corresponding changes in image-optimizer package)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
-    "name": "spatie/laravel-image-optimizer",
+    "name": "fatonsopa/laravel-image-optimizer",
```
Comment (freekmurze) [inline composer.json]: This doesn't seem right. Could you use our package?

File: src/OptimizerChainFactory.php [hunk 2]
```diff
@@ -40,13 +40,20 @@ protected static function getLogger($config): LoggerInterface
...
-            })
-            ->toArray();
+          ->mapWithKeys(function (array $options, string $optimizerClass) use($config) {
+              if (! is_a($optimizerClass, Optimizer::class, true)) {
+                  throw InvalidConfiguration::notAnOptimizer($optimizerClass);
+              }
+
+              // Initialize optimizer class
+              $newOptimizerClass = new $optimizerClass();
+              if(isset($config['binary_path']) && $config['binary_path']) {
```
Comment (freekmurze) [inline src/OptimizerChainFactory.php]: Move this to its dedicated method. That method should return an empty string if the config option is not set.

Comment (fatonsopa) [reply src/OptimizerChainFactory.php]: Just addressed this issue with the newest commit. 

https://github.com/spatie/laravel-image-optimizer/pull/49/commits/c8245f5fc47fb2cb286c06ec0de4ce0f3f87640d

Can you please check?

Thanks:

**Comment by freekmurze** [comment]
Thanks!

**Comment by fatonsopa** [comment]
You're welcome. Thanks to you!

**Comment by fatonsopa** [comment]
Hi Freek,

Would you please consider this PR considerin binary path too https://github.com/spatie/image-optimizer/pull/57 

Moreover, will you create a new release for both packages or we have to use the dev-master version?

Thakns,
Faton

**Pull request merged**